### PR TITLE
Modularization

### PR DESCRIPTION
**Pull request by kdk96**
- project divided into modules
- migrating to dagger
- edit gitignore

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -59,75 +63,83 @@ dependencies {
...
+     * Android Test
      */
-    androidTestImplementation androidTestDependencies.runner
-    androidTestImplementation androidTestDependencies.espresso
+    androidTestImplementation androidTestDeps.runner
+    androidTestImplementation androidTestDeps.espresso
+
+
+    implementation project(':common')
+
```
Comment (Hukumister) [inline app/build.gradle:128]: Why so big space is here?

File: app/src/main/java/ru/coderedwolf/wordlearn/di/DatabaseModule.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import dagger.Provides
+import ru.coderedwolf.wordlearn.database.DataBase
+import ru.coderedwolf.wordlearn.database.mapper.CategoryMapper
+import ru.coderedwolf.wordlearn.database.mapper.PhraseMapper
+import ru.coderedwolf.wordlearn.database.mapper.PhraseTopicMapper
+import ru.coderedwolf.wordlearn.database.mapper.WordMapper
+import javax.inject.Singleton
+
+@Module
+object DatabaseModule {
```
Comment (Hukumister) [inline app/src/main/java/ru/coderedwolf/wordlearn/di/DatabaseModule.kt:15]: Are you sure, that we'll have a profit if we separete DataBase on many Dao dependencies?

**Review by Hukumister — APPROVED** [review]

File: app/build.gradle (hunk 1 above)
Comment (kdk96) [reply app/build.gradle:128]: Cause I was so tired to format code

File: app/src/main/java/ru/coderedwolf/wordlearn/di/DatabaseModule.kt (hunk 2 above)
Comment (kdk96) [reply app/src/main/java/ru/coderedwolf/wordlearn/di/DatabaseModule.kt:15]: Yes, I am, cause we will provide only needed daos

File: app/build.gradle (hunk 1 above)
Comment (Hukumister) [reply app/build.gradle:128]: Format please before merge)

**Pull request merged**